consensus/src/dag/dag_store.rs

```diff
@@ -226,46 +215,23 @@ impl Dag {
         self.nodes_by_digest.get(digest).cloned()
     }
 
-    pub fn get_unlinked_nodes_for_new_round(
+    pub fn get_strong_links_for_round(
         &self,
+        round: Round,
         validator_verifier: &ValidatorVerifier,
     ) -> Option<Vec<NodeMetadata>> {
-        let current_round = self.highest_round();
-        let strong_link_authors =
-            self.highest_unlinked_nodes_by_author
-                .iter()
-                .filter_map(|maybe_node| {
-                    maybe_node.as_ref().and_then(|node| {
-                        if node.metadata.round == current_round {
-                            Some(&node.metadata.author)
-                        } else {
-                            None
-                        }
-                    })
-                });
+        let all_nodes_in_round = self.nodes_by_round.get(&round)?.iter().flatten();
         if validator_verifier
-            .check_voting_power(strong_link_authors)
+            .check_voting_power(all_nodes_in_round.clone().map(|node| &node.metadata.author))
```

Comment:
nit: can we clone the vec<author> instead? Not sure which is better.

Reply:
it's a flat map type, not the vector, I was trying to save the iter().flattern() piece